src/brisk.jl

```diff
@@ -0,0 +1,109 @@
+"""
+```
+brisk_params = BRISK([pattern_scale = 1.0])
+```
+
+| Argument | Type | Description |
+|----------|------|-------------|
+| `pattern_scale` | `Float64` | Scaling factor for the sampling window | 
+"""
+type BRISK{S, T, O} <: Params
+    threshold::Float64
+    octaves::Int
+    pattern_scale::Float64
+    pattern_table::Vector{Vector{S}}
+    smoothing_table::Vector{T}
+    orientation_weights::Vector{O}
+    short_pairs::Vector{O}
+    long_pairs::Vector{O}
+end
+
+typealias OrientationPair Vector{Int}
+typealias SamplePair Vector{Float64}
+
+function BRISK(; threshold::Float64 = 0.25, octaves::Int = 4, pattern_scale = 1.0)
+    pattern_table, smoothing_table = _brisk_tables(pattern_scale)
+    orientation_weights = OrientationPair[]
+    short_pairs = OrientationPair[]
+    long_pairs = OrientationPair[]
+    dminsq = (brisk_dmin * pattern_scale) ^ 2
+    dmaxsq = (brisk_dmax * pattern_scale) ^ 2
+    for i in 2:brisk_points
+        for j in 1:i - 1
+            dy, dx = pattern_table[1][j] - pattern_table[1][i]
+            norm = dy ^ 2 + dx ^ 2
+            if norm > dminsq
+                push!(long_pairs, OrientationPair([j, i]))
+                push!(orientation_weights, OrientationPair([round(Int, dy * 4096 / norm), round(Int, dx * 4096 / norm)]))
+            elseif norm < dmaxsq
+                push!(short_pairs, OrientationPair([j, i]))
+            end
+        end
+    end
+    BRISK(threshold, octaves, pattern_scale, pattern_table, smoothing_table, orientation_weights, short_pairs, long_pairs)
+end
+
+function _brisk_orientation{T<:Gray}(int_img::AbstractArray{T, 2}, keypoint::Keypoint, pattern::Array{SamplePair}, 
+                                        orientation_weights::Array{OrientationPair}, sigmas::Array{Float64}, long_pairs::Array{OrientationPair})
+    direction_sum_y = 0.0
+    direction_sum_x = 0.0
+    for (i, o) in enumerate(long_pairs)
+        offset_1 = pattern[o[1]]
+        offset_2 = pattern[o[2]]
+        intensity_diff = ImageFeatures._freak_mean_intensity(int_img, keypoint, offset_1, sigmas[o[1]]) - ImageFeatures._freak_mean_intensity(int_img, keypoint, offset_2, sigmas[o[2]])
+        direction_sum_y += orientation_weights[i][1] * intensity_diff / 4096
```

Comment:
I was trying to reduce storage in the `orientation_weights` vector by using Int values instead of Float. Now that I think of it, Int by default takes the same storage as Float right? I will have to explicitly specify Int16 right?